lib/rom/sql/function.rb

```diff
@@ -26,6 +26,21 @@ def is(other)
         ::Sequel::SQL::BooleanExpression.new(:'=', func, other)
       end
 
+      # Convert an expression result to another data type
+      #
+      # @example
+      #   users.select { bool::cast(json_data.get_text('activated'), :boolean).as(:activated) }
```

Comment:
We should be able to infer cast types based on type spec, so this could be simplified to be:

``` ruby
users.select { bool::json_data.get_text('activated').as(:activated) }
```

WDYT?